src/diagnostics/build.rs

```diff
@@ -9,11 +9,11 @@ use super::{Diagnostic, DiagnosticCode};
 
 pub fn collect_build_diagnostics(
     all_diagnostics: &DashMap<Arc<Url>, Vec<Diagnostic>>,
-    workspace: &Workspace,
+    workspace: &mut Workspace,
```

Comment:
Mutating is no problem here, but note that the changes made to the workspace are not persisted. The general idea is that the `Workspace` on the main thread (the one that dispatches the LSP messages) is mutable. Change notifications are handled synchronously, while the LSP features are handled asynchronously so each request gets its own `Workspace` copy (which is relatively cheap to do). This way, requests that take longer to execute do not block the main thread.